### PR TITLE
Replace chrono with time

### PR DESCRIPTION
**Pull request by Atrox**
Replaces `chrono` with `time`.
Thank you for this crate btw :).

fixes https://github.com/rustyhorde/vergen/issues/94

### CODE REVIEW

File: src/feature/git.rs [hunk 1]
```diff
@@ -222,20 +222,22 @@ where
...
                         crate::TimeZone::Local => {
-                            add_config_entries(config, git_config, &offset.with_timezone(&Local));
+                            add_config_entries(
+                                config,
+                                git_config,
+                                &timestamp.to_offset(
+                                    UtcOffset::current_local_offset()
+                                        .expect("unable to get local offset"),
+                                ),
+                            );
```
Comment (Atrox) [inline src/feature/git.rs:240]: FYI: I'm not sure if this works the same as before for local times.

**Comment by CraZySacX** [comment]
This broke some tests (build_local_timezone).   Appears that the `time` crate can't determine the local offset for some reason on linux and mac at least for the current msrv of 1.52.   I'm pulling the branch locally to see if I can reproduce.

**Comment by CraZySacX** [comment]
Looking at the crate, seeing this test does not instill much confidence: https://github.com/time-rs/time/blob/main/tests/integration/offset_date_time.rs#L14-L22

Also, local only works on unix if the running process can be guaranteed to be single threaded.   I'm not sure this crate will meet our needs here.  (https://github.com/time-rs/time/blob/main/src/sys/local_offset_at/unix.rs#L102-L118)

**Comment by Atrox** [comment]
Oh, that is totally fair. I didn't know about the local time limitations cause I don't use that at all. :)

**Comment by Atrox** [comment]
@CraZySacX I did take a closer look. It seems like chrono doesn't do much better there. `chrono` just segfaults ([RUSTSEC-2020-0159](https://github.com/rustsec/advisory-db/blob/main/crates/chrono/RUSTSEC-2020-0159.md)) in that case, `time` handles that case "gracefully" at least. That's why a lot of migrations are happening from chrono -> time from what I can tell.

https://github.com/chronotope/chrono/blob/f6bd567bb677262645c1fc3131c8c1071cd77ec3/src/sys/unix.rs#L84

**Comment by jasonozias** [comment]
I'm good with re-opening this PR if you would like to give it a go.   I would consider putting the entire local time functionality behind a feature, and only enabling the unsafe feature on the `time` crate if the user enables the "localtime" feature in this crate.

**Pull request closed without merging**